### ISSUE TITLE
onClick(tag, ...) still called after GameObj is destroyed

### ISSUE THREAD

**Issue by pjoe** [replit/kaboom#717]
After calling `GameObj.destroy()` on an instance, I'm still getting events (`onClick("someTag", ...`) for that object.

Would expect that when I destroy a GameObj that it can no longer generate events.

See replit: https://replit.com/@pjoePanda/kaboom3000-destroy-bug

**Comment by DryCreations** [comment]
It might work to modify the events in the area component a bit:

```js
onClick(this: GameObj<AreaComp>, f: () => void): EventController {
  const e = app.onMousePress("left", () => {
    if (this.isHovering()) {
      f()
    }
  })
  events.push(e)
  return e
},
```

where `events` is defined earlier in the `area` function:

```js
function area(opt: AreaCompOpt = {}): AreaComp {
  const colliding = {}
  const collidingThisFrame = new Set()
  const events = []
```

then adding a `destroy` method to take care of the event controllers tied to this component:

```js
destroy() {
  events.forEach((e)=>e.cancel())
},
```

We would probably want to add this to ALL of the events inside of `area`, but it seems like it resolves the issue you're describing.

**Comment by slmjkdbtl** [comment]
Fixed in 3000.0.1! https://github.com/replit/kaboom/commit/6b01347ee721f1daea92054877d68ddacd019787

**Comment by pjoe** [comment]
Thanks, can confirm this fixes the issue for me :tada:
